pkg/transparentproxy/transparentproxy_v2.go

```diff
@@ -198,42 +188,72 @@ func (tp *TransparentProxyV2) Setup(tpConfig *config.TransparentProxyConfig) (st
 	return Setup(cfg)
 }
 
-func parseExcludePortsForUIDs(excludeOutboundPortsForUIDs []string, protocol string) ([]config.UIDsToPorts, error) {
+func parseExcludePortsForUIDs(excludeOutboundPortsForUIDs []string) ([]config.UIDsToPorts, error) {
 	var uidsToPorts []config.UIDsToPorts
 	for _, excludePort := range excludeOutboundPortsForUIDs {
 		parts := strings.Split(excludePort, ":")
-		if len(parts) != 2 {
-			return nil, errors.New("value contains too many \":\" - format for excluding ports by UIDs ports:uids")
+		if len(parts) == 0 || len(parts) > 3 {
+			return nil, fmt.Errorf("value: '%s' is invalid - format for excluding ports by UIDs <protocol:>?<ports:>?uids", excludePort)
+		}
+		var portValuesOrRange string
+		var protocolOpts string
+		var uidValuesOrRange string
+		switch len(parts) {
+		case 1:
+			protocolOpts = "*"
+			portValuesOrRange = "*"
+			uidValuesOrRange = parts[0]
+		case 2:
+			protocolOpts = "*"
+			portValuesOrRange = parts[0]
+			uidValuesOrRange = parts[1]
+		case 3:
+			protocolOpts = parts[0]
+			portValuesOrRange = parts[1]
+			uidValuesOrRange = parts[2]
+		}
+		if uidValuesOrRange == "*" {
+			return nil, errors.New("can't use wildcard '*' for uids")
+		}
+		if portValuesOrRange == "*" || portValuesOrRange == "" {
+			portValuesOrRange = "0-65535"
```

Comment:
I'm not sure we should actually redirect or allow to use port `0` as it's reserved and using it could have unexpected behavior. I.e. Linux treats it as a wildcard port and when used, kernel will draw, free port and use it instead.

Reply:
100%